### PR TITLE
Better client_url handling

### PR DESCRIPTION
**Pull request by dinhlongviolin1**

### CODE REVIEW

File: taipy/gui/server.py [hunk 1]
```diff
@@ -337,12 +337,14 @@ def run(
...
             else:
-                _TaipyLogger._get_logger().info(f" * Server reloaded on http://{host_value}:{port}")
-            _TaipyLogger._get_logger().info(f" * Application is accessible at {client_url}")
+                _TaipyLogger._get_logger().info(f" * Server reloaded on {server_url}")
+            if client_url is not None:
+                client_url = client_url.format(port=port)
+                _TaipyLogger._get_logger().info(f" * Application is accessible at {client_url}")
         if not is_running_from_reloader() and self._gui._get_config("run_browser", False):
-            webbrowser.open(client_url, new=2)
+            webbrowser.open(client_url or server_url, new=2)
```
Comment (FabienLelaquais) [inline taipy/gui/server.py:347]: Yeah this makes sense.

File: taipy/gui/server.py [hunk 2]
```diff
@@ -337,12 +337,14 @@ def run(
             log = logging.getLogger("werkzeug")
             log.disabled = True
             if not is_running_from_reloader():
```
Comment (FabienLelaquais) [inline taipy/gui/server.py:339]: Why the test?

**Review by FabienLelaquais — DISMISSED** [review]
👍

**Review by FredLL-Avaiga — DISMISSED** [review]
👍

File: taipy/gui/server.py (hunk 2 above)
Comment (dinhlongviolin1) [reply taipy/gui/server.py:339]: different messages as you can see. The test is for the reloader during dev mode

**Review by FredLL-Avaiga — APPROVED** [review]

**Review by FabienLelaquais — APPROVED** [review]
👍

**Pull request merged**